Author: Jordi Mallach <jordi@debian.org>
Description: Check for and include libavutil.h to avoid unresolved symbols.
 The a52 plugin uses av_free(), but does not check for libavutil
 availability. Patch configure.in to check for the module, and include
 libavutil.h.
 .
 The patch needs to be extended to do proper checks for avutil header files.
Forwarded: no

Index: alsa-plugins-1.0.25/configure.in
===================================================================
--- alsa-plugins-1.0.25.orig/configure.in	2012-01-25 08:57:07.000000000 +0100
+++ alsa-plugins-1.0.25/configure.in	2012-02-11 22:22:41.439521425 +0100
@@ -67,7 +67,7 @@
       AS_HELP_STRING([--disable-avcodec], [Don't build plugins depending on avcodec (a52)]))
 
 if test "x$enable_avcodec" != "xno"; then
-  PKG_CHECK_MODULES(AVCODEC, [libavcodec], [HAVE_AVCODEC=yes], [HAVE_AVCODEC=no])
+  PKG_CHECK_MODULES(AVCODEC, [libavcodec libavutil], [HAVE_AVCODEC=yes], [HAVE_AVCODEC=no])
 fi
 
 if test "x$HAVE_AVCODEC" = "xno"; then
Index: alsa-plugins-1.0.25/a52/pcm_a52.c
===================================================================
--- alsa-plugins-1.0.25.orig/a52/pcm_a52.c	2012-01-25 08:57:07.000000000 +0100
+++ alsa-plugins-1.0.25/a52/pcm_a52.c	2012-02-11 22:28:58.219502171 +0100
@@ -26,6 +26,7 @@
 #include <alsa/pcm_external.h>
 #include <alsa/pcm_plugin.h>
 #include AVCODEC_HEADER
+#include <libavutil/avutil.h>
 
 struct a52_ctx {
 	snd_pcm_ioplug_t io;
